File: leetcode/2260.cpp
class Solution
{
public:
    int minimumCardPickup(vector<int> &cards)
    {
        unordered_map<int, int> seen_index;
        int ans = -1;
        for (int i = 0, n = cards.size(); i < n; ++i)
        {
            auto &card = cards[i];
            if (seen_index.find(card) != seen_index.end())
            {
                int cur_ans = i - seen_index[card] + 1;
                if (ans == -1 || cur_ans < ans)
                {
                    ans = cur_ans;
                }
            }
            seen_index[card] = i;
        }
        return ans;
    }
};

File: leetcode/2260.py
class Solution:
    def minimumCardPickup(self, cards: List[int]) -> int:
        ans, seen = -1, {}
        for index, card in enumerate(cards):
            if card in seen:
                cur_ans = index - seen[card] + 1
                if ans == -1 or ans > cur_ans:
                    ans = cur_ans
            seen[card] = index
        return ans
